### ISSUE TITLE
Add a "Concept" article on Spin Templates

### ISSUE THREAD

**Issue by macolso** [fermyon/developer#363]
We use templates frequently in our developer documentation. We have advanced "Reference" material on how to create templates, but are missing a "Concept" article that explains what Spin templates are in the first place, why they're useful, and when someone would want to use a pre-defined template. 

A "Concept" article is oriented on understanding and must educate the reader on a specific topic using discursive explanation.

**Comment by tpmccallum** [comment]
While collaborating on the "managing templates" page some info about how templates and SDKs overlap became apparent.
I am pasting in the content (that was removed to simplify the "managing templates" page) here in the hope that it can be used in a more conceptual context.
---
START

#### Installing as Part of a Specific Language SDK

As mentioned in the [Introducing the Spin JavaScript and TypeScript SDK](https://www.fermyon.com/blog/spin-js-sdk) blog article, you can install a template relating to a specific language SDK. For example, the template relating specifically to the [Spin JS SDK](https://github.com/fermyon/spin-js-sdk) can be installed using the following command:



```bash
spin templates install --git https://github.com/fermyon/spin-js-sdk
```

> Note: Building Spin components in JavaScript is experimental and requires more than just the template. Please see the [Javascript Components](https://developer.fermyon.com/spin/javascript-components) section of the documentation for more information. Similarly, if interested in building Spin components in Python (also currently experimental) see the ([spin-python-sdk](https://github.com/fermyon/spin-python-sdk) Git repository, [blog article](https://www.fermyon.com/blog/spin-python-sdk) and [technical documentation](https://developer.fermyon.com/spin/python-components) for more information).

---
END

In addition we would start to also talk about new[er] languages as they become part of the mainstream docs i.e. .NET and BF etc. etc.

> [WebAssembly for .NET Developers: Introducing the Spin .NET SDK](https://www.fermyon.com/blog/webassembly-for-dotnet-developers-spin-sdk-intro)

**Comment by quyumkehinde** [comment]
@macolso does this article conver? https://www.fermyon.com/blog/component-reuse

**Comment by itowlson** [comment]
@quyumkehinde That's a blog post (and is quite old).  Reopening because this remains an issue for the Spin docs site.

**Comment by macolso** [comment]
I agree with @itowlson—this is still needed. For example, it’s not immediately clear how templates are versioned or how they relate to CLI and SDK versions. I also think it’s important to clarify the distinction between templates and components. There are many other questions that could be addressed in an article like this.
